iris-mpc-cpu/src/hnsw/searcher.rs

```diff
@@ -1227,10 +1290,49 @@ impl HnswSearcher {
         graph: &GraphMem<V::VectorRef>,
         query: &V::QueryRef,
         insertion_layer: usize,
-    ) -> Result<(Vec<SortedNeighborhoodV<V>>, bool)> {
+    ) -> Result<(Vec<SortedNeighborhoodV<V>>, SetEntryPoint)> {
         let mut links = vec![];
 
-        let (mut W, n_layers) = self.search_init(store, graph, query).await?;
+        let (mut W, n_layers, set_ep) = match self.params.top_layer_mode {
```

Comment:
My feeling is that this entire logic should go into `search_init` and  `search_to_insert` should be as similar as possible to the version on `dev`.

Reply:
the set entry point logic throws a wrench i this but i was thinking of moving that inside of `search_init` and then ignoring `set_ep` during `search`.